### PR TITLE
BI-6176 Change AvroSerializer to use SpecificRecords

### PR DESCRIPTION
**Pull request by markpit**
Using the AvroSerializer from https://github.com/companieshouse/kafka-models Readme to use SpecificRecord

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java [hunk 1]
```diff
@@ -24,9 +23,6 @@
 
     private CHKafkaProducer producer;
 
-    @Autowired
-    private Schema schema;
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java:28]: Presumably there's a schema or getSchema bean declared somewhere, is that bean used anywhere else now that it is no longer used here?

File: src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java [hunk 2]
```diff
@@ -24,9 +23,6 @@
 
     private CHKafkaProducer producer;
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java]: My intellij is flaggin that these 3 fields, logger, avroSerializer and producer can all be marked as final

File: src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java (hunk 1 above)
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java:28]: you can now get the schema from the model class  - see https://github.com/companieshouse/chips-rest-interfaces-consumer/pull/29

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImpl.java:28]: i see what you mean , we can remove the config for this as well

File: src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/AvroSerializer.java [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
-    public byte[] serialize(ChipsRestInterfacesSend message, Schema schema) throws IOException {
-        GenericDatumWriter<ChipsRestInterfacesSend> datumWriter = new GenericDatumWriter<>();
+    byte[] serialize(ChipsRestInterfacesSend data) throws IOException {
+        DatumWriter<ChipsRestInterfacesSend> datumWriter = new SpecificDatumWriter<>();
 
-        try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
-            Encoder encoder = EncoderFactory.get().binaryEncoder(stream, null);
-            datumWriter.setSchema(schema);
-            datumWriter.write(message, encoder);
+        try(ByteArrayOutputStream out = new ByteArrayOutputStream()) {
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/AvroSerializer.java]: missing space after try

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/AvroSerializer.java]: should be updated now

**Review by ICoxCH — APPROVED** [review]

**Pull request merged**